kratos/utilities/parallel_utilities.h

```diff
@@ -94,6 +94,12 @@ class KRATOS_API(KRATOS_CORE) ParallelUtilities
      */
     [[nodiscard]] static int GetNumProcs();
 
+    /**
+     * @brief Wrapper for omp_in_parallel()
+     * @return Maximum number of OpenMP threads that will be used in parallel regions.
+     */
+    [[nodiscard]] static int IsInParallel();
```

Comment:
this was deliberately not added so far since there is no good solution for Cxx-based parallelisation. I did lots of testing on this, but couldnt get it to work
IIRC the main issue was that the main-thread contributes also in the parallel loop, thus its not as simple as to just check if the current thread-id is different from the main-thread id

Reply:
my reasoning for not adding it was that if it doesnt work generally it shouldnt be in the ParallelUtils

Its sth OpenMP specific, thus should remain there until we figure it out

Reply:
my opinion, feel free to ignore :upside_down_face:

Reply:
Of course, I don't have time to invest on that. Anyway no one is using the C++11 parallel and we should support the C++17 which was actually game changer.